src/connector/src/sink/doris_connector.rs

```diff
@@ -0,0 +1,412 @@
+use bytes::{BufMut, Bytes, BytesMut};
+use http::request::Builder;
+use hyper::body::{Body, Sender};
+use hyper::client::HttpConnector;
+use hyper::{body, Client, Request, StatusCode};
+use hyper_tls::HttpsConnector;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use tokio::task::JoinHandle;
+
+use super::{Result, SinkError};
+
+const BUFFER_SIZE: usize = 64 * 1024;
+const MIN_CHUNK_SIZE: usize = BUFFER_SIZE - 1024;
+pub struct DorisInsertClient {
+    url: String,
+    header: HashMap<String, String>,
+    sender: Option<Sender>,
+}
+impl DorisInsertClient {
+    pub fn new(url: String, db: String, table: String) -> Self {
+        let url = format!("{}/api/{}/{}/_stream_load", url, db, table);
+        Self {
+            url,
+            header: HashMap::default(),
+            sender: None,
+        }
+    }
+
+    pub fn set_url(mut self, url: String) -> Self {
+        self.url = url;
+        self
+    }
+
+    pub fn add_common_header(mut self) -> Self {
+        self.header
+            .insert("expect".to_string(), "100-continue".to_string());
+        self
+    }
+
+    pub fn set_label(mut self, label: String) -> Self {
+        self.header.insert("label".to_string(), label);
+        self
+    }
+
+    pub fn add_hidden_column(mut self) -> Self {
+        self.header.insert(
+            "hidden_columns".to_string(),
+            "__DORIS_DELETE_SIGN__".to_string(),
+        );
+        self
+    }
+
+    pub fn enable_2_pc(mut self) -> Self {
+        self.header
+            .insert("two_phase_commit".to_string(), "true".to_string());
+        self
+    }
+
+    pub fn set_user_password(mut self, user: String, password: String) -> Self {
+        let auth = format!(
+            "Basic {}",
+            general_purpose::STANDARD.encode(format!("{}:{}", user, password))
+        );
+        self.header.insert("Authorization".to_string(), auth);
+        self
+    }
+
+    pub fn set_txn_id(mut self, txn_id: i64) -> Self {
+        self.header
+            .insert("txn_operation".to_string(), txn_id.to_string());
+        self
+    }
+
+    pub fn add_commit(mut self) -> Self {
+        self.header
+            .insert("txn_operation".to_string(), "commit".to_string());
+        self
+    }
+
+    pub fn add_abort(mut self) -> Self {
+        self.header
+            .insert("txn_operation".to_string(), "abort".to_string());
+        self
+    }
+
+    pub fn set_properties(mut self, properties: HashMap<String, String>) -> Self {
+        self.header.extend(properties);
+        self
+    }
+
+    fn build_request_and_client(
+        &self,
+        uri: String,
+    ) -> (Builder, Client<HttpsConnector<HttpConnector>>) {
+        let mut builder = Request::put(uri);
+        for (k, v) in &self.header {
+            builder = builder.header(k, v);
+        }
+
+        let connector = HttpsConnector::new();
+        let client = Client::builder()
+            .pool_idle_timeout(Duration::from_secs(10))
```

Comment:
There are many `Duration::from_secs(10)` in this file. I think it is better to store these magic number in CONSTs at the beginning of this file so that we need what different timeout is.